cds-snc/notification-utils
# Add formatter pre-commit hook

# Summary | Résumé

This PR adds a Ruff formatter `pre-commit` hook. Whenever `git commit` is performed, the hook will run the formatter on the staged files. Once complete just stage the formatting changes and continue with the commit.

**Note:**
If you do your committing outside of the vscode integrated terminal, then in order for this hook to work you need to:
1.  install the `pre-commit` package into your local python environment 
2. Run `pre-commit install` while in the repo 
3. Run `pre-commit run` to test if it works

![image](https://github.com/user-attachments/assets/0f7815dc-12cb-4477-8d9f-7363ffcce4ef)

# Test instructions | Instructions pour tester la modification

## Testing python file changes
1. Build and open the dev container
3. Edit a python file, rearranging the import order 
5. Commit the file
6. Note that the two pre-commit hook actions, (ruff, ruff-format) are run.
7. Do a `git status` and inspect the formatting changes

![image](https://github.com/user-attachments/assets/5703d819-71a8-4037-a809-d5add514d0bb)

## Testing non-python file changes
1. Make a small edit to the `makefile` 
2. Commit the change
3. Note that both the `ruff` and `ruff-format` hooks are skipped, and the commit command completes

![image](https://github.com/user-attachments/assets/edac19aa-01a5-49cf-8c96-cb7cf9910f4b)


# Reviewer checklist | Liste de vérification du réviseur


> ⚠ If boxes cannot be checked off before merging the PR, they should be moved to the "Release Instructions" section with appropriate steps required to verify before release. For example, changes to celery code may require tests on staging to verify that performance has not been affected.
